include/openssl/store.h

```diff
@@ -84,6 +84,11 @@ int OSSL_STORE_vctrl(OSSL_STORE_CTX *ctx, int cmd, va_list args);
  */
 OSSL_STORE_INFO *OSSL_STORE_load(OSSL_STORE_CTX *ctx);
 
+OSSL_STORE_INFO *OSSL_STORE_load_with_libctx(OSSL_STORE_CTX *ctx,
+                                             OPENSSL_CTX *libctx,
+                                             const char *propq);
+
```

Comment:
This doesn't make much sense, really, if you consider OSSL_STORE for providers, and that scheme based methods will be fetched by `OSSL_STORE_open()`.  What I did was this:

```
+OSSL_STORE_CTX *OSSL_STORE_open_with_libctx(OPENSSL_CTX *libctx,
+                                   const char *uri, const char *properties,
+                                   const UI_METHOD *ui_method, void *ui_data,
+                                   OSSL_STORE_post_process_info_fn post_process,
+                                   void *post_process_data);
```

There should obviously be a corresponding `OSSL_STORE_open_with_libctx_fn` that takes a libctx and propq and saves them away in the loader context.

Reply:
Not sure I understand.. Do you mean the function signature is this??

```
OSSL_STORE_CTX *OSSL_STORE_open_with_libctx(OPENSSL_CTX *libctx,
                                            const char *uri, const char *properties,
                                            const UI_METHOD *ui_method, void *ui_data,
                                            OSSL_STORE_post_process_info_fn post_process,
                                            void *post_process_data,
                                            OSSL_STORE_open_with_libctx_fn);
```

Reply:
And will that work in PEM_read_bio_PrivateKey_ex
Which does an OSSL_STORE_attach() followed by OSSL_STORE_load()

Reply:
> Not sure I understand.. Do you mean the function signature is this??

No.  Why would I pass the method function pointer?  It should be in the `OSSL_STORE_LOADER` that it gets from the URI scheme.

> And will that work in PEM_read_bio_PrivateKey_ex
> Which does an OSSL_STORE_attach() followed by OSSL_STORE_load()

`OSSL_STORE_attach()` already takes a libctx and propq (and propagates them to the OSSL_STORE_LOADER` `attach` method, something you removed?????), so I fail to see the issue.

Reply:
So basically, what I'm trying to say is that I'd prefer specifying the libctx and propq on init (open or attach) rather than on load.  That corresponds pretty much with how we operate elsewhere.

Reply:
Fair enough..
I think the original problem I had was that the libctx and propq are sitting in a object that is not passed to other lower level objects. As they dont know who the parent is they also dont know what the libctx and propq are either. I will have to code a bit to try to remember what exactly was the problem (this was done many weeks ago).